docs/guides/es2k/compiling-p4-programs.md

```diff
@@ -2,134 +2,107 @@
-```
-
-Verify the compiler version number:
-
-```bash
-[root@host ~] # p4c-pna-xxp --version
-p4c-pna-xxp
-Version 3.0.70.7
 ```
 
 The compiler should now be ready for use.
 
-## 3. P4 Reference Files
+## 3. Build Reference P4 programs
 
-The `/usr/share/mev_reference_p4_files/` directory contains a number of sample
-P4 programs.
+The `p4-programs` directory in the SDK folder contains a number of sample
+P4 programs. The SDK is contained in a tar file intel-ipu-sdk-source-code-<$VERSION>.tgz.
+Extract the p4-programs from the tarball.
 
 ```bash
-[root@host ~]# ls -l /usr/share/mev_reference_p4_files/
-total 28
--rw-r--r--. 1 root root   79 Jun  8 08:06 CONTENTS
-drwxr-xr-x. 2 root root 4096 Jun  8 03:21 linux_networking
--rw-r--r--. 1 root root 1104 Jun  8 08:06 mev_sample.conf
-drwxr-xr-x. 2 root root 4096 Jun  8 03:21 pna_connection_track
-drwxr-xr-x. 2 root root 4096 Jun  8 03:21 simple_l2_demo
-drwxr-xr-x. 2 root root 4096 Jun  8 03:21 simple_l2_mod_demo
-drwxr-xr-x. 2 root root 4096 Jun  8 03:21 simple_l3_l4_pna
+[root@host ~]# ls -l $INTEL-IPU-SDK-<VERSION>/tools/pipeline/p4-programs
+total 1100
+-rw-r--r--.  1 user user     79 Dec 13 15:54 CONTENTS
+drwxr-xr-x. 10 user user   4096 Dec 13 15:54 fxp-cxp-features
+drwxr-xr-x. 10 user user   4096 Dec 13 15:54 fxp-net-scenarios
+drwxr-xr-x.  7 user user    106 Dec 13 15:54 layer-2-forwarding
+drwxr-xr-x.  8 user user    116 Dec 13 15:54 layer-3-forwarding
+-rw-r--r--.  1 user user  81529 Dec 23 03:09 Makefile
+-rw-r--r--.  1 user user   1494 Dec 13 15:54 mev_sample.conf
+drwxr-xr-x.  2 user user    131 Dec 13 15:54 testsuite
+
 ```
 
 Each P4 program has its own directory and an accompanying README file that
 provides instructions on how to configure the IPU pipeline.
 
-## 4. Compiling a P4 Program
+## 3.1 Compiling a P4 Program
 
-Use `p4c-pna-xxp` to compile a P4 program. We will be using one of the reference
-programs mentioned above.
+Use `p4c` i.e the compiler driver to compile and build packages. 
+We will be using one of the reference programs mentioned above as an example -
+p4-programs/layer-3-forwarding/l3-fwd_p2p. The Makefile contains 
+the exact commands to  build the artifacts and .pkg for all the reference programs.
 
 ```bash
-# Set environment variables
-export LD_LIBRARY_PATH=/usr/lib:/usr/lib64:/usr/local/lib:/usr/local/lib64:$LD_LIBRARY_PATH
-export OUTPUT_DIR=/usr/share/mev_reference_p4_files/simple_l3_l4_pna
-
-# Compile p4 program
-p4c-pna-xxp -I/usr/lib -I/usr/share/p4c/p4include -I/usr/share/p4c/idpf-lib \
-            $OUTPUT_DIR/simple_l3_l4_pna.p4 -o $OUTPUT_DIR/simple_l3_l4_pna.s \
-            --p4runtime-files $OUTPUT_DIR/simple_l3_l4_pna.p4info.txt \
-            --context $OUTPUT_DIR/simple_l3_l4_pna.context.json \
-            --bfrt $OUTPUT_DIR/simple_l3_l4_pna.bf-rt.json
-```
-
-The compiler will generate the following files:
-
-- simple_l3_l4_pna.p4info.txt
-- simple_l3_l4_pna.bf-rt.json
-- simple_l3_l4_pna.context.json
-- simple_l3_l4_pna.s
+# 
```

Comment:
1. If you view this section in rendered form (click the ellipsis in the heading and select View File), I think you'll find it confusing. There's undifferentiated text mixed in with command lines, and the command lines are inconsistent. (Some are preceded by prompts; some aren't.)
2. I think this situation calls for alternating text and code blocks.
3. If you're going to show a prompt, make it _short_. I'd suggest `$ `. The pound sign makes the line look like a comment.
4. Indent the code blocks to make them subordinate to the numbered paragraphs that introduce them.

-----

For example:

1. Copy the entire `p4-programs` directory from the host to the P4Tools container.

   ```bash
   $ sudo docker cp p4-programs/ <p4Tools container id>:/opt
   ```

2. Determine the full path of the P4 compiler.

   ```bash
   $ which p4c 
   /opt/p4-tools/p4c/bin/p4c
   ```

3. Set the following environment variables in the container.

   ```bash
   $ export P4C_PATH=/opt/p4-tools/p4c/bin
   $ export CPT_PATH=/opt/p4-tools/cpt/bin
   $ export PATH=$P4C_PATH:$CPT_PATH:$PATH
   $ export LD_LIBRARY_PATH=$P4C_PATH/../lib:/usr/local/lib:$LD_LIBRARY_PATH
   $ export PREV_ARTIFACT_DIR=
   ```
   _(Did you mean to set PREV_ARTIFACT_DIR to an empty string?)_

6. Compile `l3-fwd_p2p`. All artifacts will be generated in the artifacts directory. _(Two sentences, not one with a comma splice. It's simpler, and it's grammatically correct)_

   ```bash
   $ cd /opt/p4-programs
   $ make l3-fwd_p2p
   # Compiling the /opt/p4-programs/layer-3-forwarding/l3-fwd_p2p/l3-fwd_p2p.p4 and generating the runtime files and assembly...
   /opt/p4-tools/p4c/bin/p4c --target idpf --arch pna -I/opt/p4-tools/p4c/bin/../lib -I/opt/p4-tools/p4c/bin/../share/p4c/p4include 
   -I/opt/p4-tools/p4c/bin/../share/p4c/idpf-lib --package-by-domain --p4runtime-files ./artifacts/l3-fwd_p2p/p4Info.txt --save-temps -Xp4c 
   "--Wdisable --no-pedantic --context ./artifacts/l3-fwd_p2p/context.json --bfrt ./artifacts/l3-fwd_p2p/bf-rt.json" 
   --save-temps --npic --format csr --pkg-version 1.2 --pkg-name "FXP Package" 
   -Xassembler ".cpt_ver.s" /opt/p4-programs/layer-3-forwarding/l3-fwd_p2p/l3-fwd_p2p.p4 
   -o artifacts/l3-fwd_p2p
   ```
etc.

Reply:
root@a54d354e447e was added to show user that these commands are executed in the container. 

Yes PREV_ARTIFACTS is an empty string since this is a first time build.

The long lines are actually the compiled output to show arguments being passed to the compiler.